scripts/pylib/pytest-twister-harness/src/twister_harness/device/device_adapter.py

```diff
@@ -0,0 +1,298 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import abc
+import logging
+import os
+import queue
+import re
+import shutil
+import threading
+import time
+from datetime import datetime
+from pathlib import Path
+
+from twister_harness.exceptions import (
+    TwisterHarnessException,
+    TwisterHarnessTimeoutException,
+)
+from twister_harness.twister_harness_config import DeviceConfig
+
+logger = logging.getLogger(__name__)
+
+
+class DeviceAdapter(abc.ABC):
+    """Class defines an interface for all devices."""
+
+    def __init__(self, device_config: DeviceConfig, **kwargs) -> None:
+        """
+        :param device_config: device configuration
+        :param test_name: this information is used only in header log file, it
+            shouldn't impact on adapter code logic
+        """
```

Comment:
`test_name` is not used in any adapter classes. `initialize_log_files` is called directly from dut fixture. So `**kwargs` are unnecessary in adapters for now